pkgs/by-name/st/starpu/package.nix

```
@@ -0,0 +1,106 @@
+{
+  # derivation dependencies
+  lib,
+  fetchurl,
+  stdenv,
+  writableTmpDirAsHomeHook,
+  autoreconfHook,
+
+  # starpu dependencies
+  hwloc,
+  libuuid,
+  libX11,
+  fftw,
+  fftwFloat, # Same than previous but with float precision
+  pkg-config,
+  libtool,
+  simgrid,
+  mpi,
+  cudaPackages,
+
+  # Options
+  enableSimgrid ? false,
+  enableMPI ? false,
+  enableCUDA ? false,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "starpu";
+  version = "1.4.7";
+
+  inherit enableSimgrid;
+  inherit enableMPI;
+  inherit enableCUDA;
+
+  src = fetchurl {
+    url = "https://files.inria.fr/starpu/starpu-${finalAttrs.version}/starpu-${finalAttrs.version}.tar.gz";
+    hash = "sha256-HrPfVRCJFT/m4LFyrZURhDS0qB6p6qWiw4cl0NtTsT4=";
+  };
+
+  nativeBuildInputs =
+    [
+      pkg-config
+      hwloc
+      libtool
+      writableTmpDirAsHomeHook
+      autoreconfHook
+    ]
+    ++ lib.optional finalAttrs.enableSimgrid simgrid
+    ++ lib.optional finalAttrs.enableMPI mpi
+    ++ lib.optional finalAttrs.enableCUDA cudaPackages.cudatoolkit;
+
+  buildInputs =
+    [
+      libuuid
+      libX11
+      fftw
+      fftwFloat
+      hwloc
```

```suggestion
      hwloc
      python3
```
`$out/bin/starpu_fxt_number_events_to_names.py` is a Python script, this ensures it is runnable regardless of a system Python installation.